nemo_aligner/models/nlp/gpt/megatron_gpt_regression_reward_model.py

```diff
@@ -114,6 +114,47 @@ def loss_func(output_tensor):
         return fwd_output_and_loss_func
 
     def loss_func(self, output_tensor, label_tensor):
+        loss_func_name = self.cfg.get("loss_func", "regression")
+
+        if loss_func_name == "regression":
+            return self.reg_loss_func(output_tensor, label_tensor)
+        elif loss_func_name.endswith("bt"):
+            return self.bt_loss_func(output_tensor, label_tensor)
+        else:
+            raise ValueError("only accepted values for loss_func are regression, regular_bt, margin_bt and scaled_bt")
+
+    def bt_loss_func(self, output_tensor, label_tensor):
+        """
+        label_tensor should have the value of the [0,0] index as a number indicating the prefered response as well as the strength of the preferred response
+        
+        In HelpSteer2-Preference, label is a non-negative integer between -3 and 3: -3, -2, -1 means A is preferred while 1, 2, 3 means B is preferred
+        
+        output_tensor uses the value of [:, 4] index as the reward for the chosen/rejected values, which inherits the position of the helpfulness attribute
+        """
+        label = label_tensor[0, 0]
```

Comment:
I'm a bit confused here. Could you give an example of label_tensor, its shape etc?

Reply:
done - is this clearer?

Reply:
Yes , thanks